### ISSUE TITLE
[GNOME] Transmission should use monochrome tray icon

### ISSUE THREAD

**Issue by AppleMayExist** [transmission/transmission#3312]
While I haven't used Plasma or really any other DE in a while, and therefore don't know there style, I know that on GNOME, tray icons are monochrome, and Transmission's isn't. This really isn't too important, or much of an issue, but it would be nice if it could fit in with everything else. Again, not much of an issue. Below's a screenshot, with some extensions, but it would look similar without them (just no transparent panel).
![image](https://user-images.githubusercontent.com/62101291/174207000-e919ea29-71cb-48e8-bb9d-03f53eb0b0b5.png)

**Comment by GaryElshaw** [comment]
Could you please try the below 'transmission.ico' file in either the app or your src when/if you build it? It's either this or another file being used as the basis for that icon.

![transmission ico](https://user-images.githubusercontent.com/69029666/175371444-e0c9de35-4e9c-45b8-9af3-7ff64008a553.png)

**Comment by nevack** [comment]
> Could you please try the below 'transmission.ico' file in either the app or your src when/if you build it? It's either this or another file being used as the basis for that icon.
> 
> ![transmission ico](https://user-images.githubusercontent.com/69029666/175371444-e0c9de35-4e9c-45b8-9af3-7ff64008a553.png)

IMO it's better to draw something really matching, like outlined silhouette of current icon.
Just making it monochrome looks pretty bad

**Comment by GaryElshaw** [comment]
Yes, i'm just trying to ascertain whether the tray is using that icon or the 48x48 transmission.png

**Comment by nevack** [comment]
```diff
diff --git a/gtk/SystemTrayIcon.cc b/gtk/SystemTrayIcon.cc
index 349284842..2f19f3120 100644
--- a/gtk/SystemTrayIcon.cc
+++ b/gtk/SystemTrayIcon.cc
@@ -30,7 +30,7 @@ using namespace std::literals;
 namespace
 {
 
-auto const TrayIconName = Glib::ustring("transmission-tray-icon"s);
+auto const TrayIconName = Glib::ustring("transmission-symbolic"s);
 auto const AppIconName = Glib::ustring("transmission"s);
 auto const AppName = Glib::ustring("transmission-gtk"s);
```

Gives me 
![image](https://user-images.githubusercontent.com/8330119/175387314-5ff6bb59-e46d-456a-b943-eb5ba0f0ff78.png)

**Comment by ckerr** [comment]
I"ll defer to @mikedld partially because he's done more work on the GTK client the last year or so and partially because I don't know what why `transmission-symbolic` isn't loading there :smile: 

Mike I can TAL if you don't want to, but my gutcheck is you've got a better idea on this one than I do

**Comment by nevack** [comment]
I know very little about Gtk itself, but just looking at the code I see:
1. `transmission-tray-icon` is not supplied by Transmission itself, I think it's up to packages of distro to supply it OR for GTK Icon Theme creators.
2. `transmission-symbolic` is supplied, that's why it works.

I think we can try to create `transmission-tray-icon` (maybe just copy `transmission-symbolic`) and do not rely on packagers/icon-themes

I can help to test, as I'm daily driving ArchLinux on my work machine.

**Comment by GaryElshaw** [comment]
Yep. transmission-tray-icon.png

![transmission-tray-icon](https://user-images.githubusercontent.com/69029666/175474549-ae3de11d-8a41-466d-b999-28afbfdb0f67.png)

**Comment by GaryElshaw** [comment]
@AppleMayExist After having spent some more time in a Gnome linux environment, the appearance of the tray icon is due to whatever theme is being used. I'm using Manjaro, that uses a non-Adwaita theme by default. 

The non-Adwaita theme has a white tray icon, Adwaita has the coloured icon for 3.00+ builds. There are some peculiarities with both the default and the adwaita theme when it comes to Transmission's appearance. A few icons appear differently in the app itself depending on what theme is being used, sometimes ignoring Transmission's own coded icons. Hopefully someone with skills i don't possess will explore hard-coding Transmission's appearance across themes.

**Comment by mikedld** [comment]
I think it should be up to system tray implementation to decide whether to render symbolic or normal icon. The problem is, when `Gtk.StatusIcon` is used, the icon is loaded on application side, not system tray side, and application doesn't know which type of icon to load to better integrate with the system (either symbolic or normal could be correct depending on the environment and its settings). There're a couple of system tray implementations that tried to solve this, see e.g. https://blog.linuxmint.com/?p=3795. We already integrate with AppIndicator, might add support for XApp.StatusIcon as part of this ticket. Note that, regretfully, none of these solutions currently support GTK 4, so system tray icon support is disabled there.

**Comment by GaryElshaw** [comment]
Wow, i see what you mean.

`"The icon widget is rendered by the application, so the panel applet has no control over it. It can’t resize it properly, it can’t parse its icon name or switch the icon to symbolic, and depending on how the app sets its icon, when and how, we can sometimes end up with rendering issues… invisible icons, cropped icons, wrongly sized icons or icons which can’t get bigger than 16px."`
